src/MongoDB.Driver.Core/Core/Misc/DnsClientWrapper.cs

```diff
@@ -25,18 +25,20 @@ namespace MongoDB.Driver.Core.Misc
 {
     internal class DnsClientWrapper : IDnsResolver
     {
+        #region static
+        private static IDnsResolver __instance;
+        public static IDnsResolver Instance => __instance ?? (__instance = new DnsClientWrapper());
```

Comment:
I see no reason for it to be lazy. Refactor to:

    private static IDnsResolver __instance = new DnsClientWrapper();
    public static IDnsResolver Instance => __instance;

Reply:
done

Reply:
I'm curious: why not use a Lazy?

Reply:
Ah, we appear to use the instance in each branch of the if statement in `Resolve`.

Reply:
Are you looking at the right version?

This property is now different.